lib/dradis/plugins/template_service.rb

```diff
@@ -1,23 +1,50 @@
 module Dradis
   module Plugins
     class TemplateService
-      attr_accessor :logger, :template, :templates_dir
+      attr_accessor :logger, :plugin, :processor, :project, :template, :templates_dir
 
       def initialize(args = {})
         @plugin        = args.fetch(:plugin)
         @templates_dir = args[:templates_dir] || default_templates_dir
+        @project       = args.fetch(:project, nil)
       end
 
       # For a given entry, return a text blob resulting from applying the
-      # chosen template to the supplied entry.
+      # appropriate mapping to the supplied entry.
       def process_template(args = {})
         self.template = args[:template]
         data          = args[:data]
+        @processor    = plugin::FieldProcessor.new(data: data)
 
-        processor = @plugin::FieldProcessor.new(data: data)
+        if mapping && mapping.mapping_fields.any?
```

Comment:
Do you think we could simplyfy this?

We could just have #apply_mapping and #apply_mapping(mapping_fields). If the mapping_fields are nil, then we just use the default mapping within the method. We can avoid duplication that way.